mc-sema/cfgToLLVM/x86Instrs_Branches.cpp

```diff
@@ -43,85 +43,46 @@
 
 using namespace llvm;
 
+template <int width>
 static InstTransResult doRet(BasicBlock *b) {
   //do a read from the location pointed to by ESP
 
-  Value *rESP = R_READ<32>(b, X86::ESP);
-  Value *nESP =
-
-  BinaryOperator::CreateAdd(rESP, CONST_V<32>(b, 4), "", b);
+  TASSERT(width == 32 || width == 64, "Invalid reg width for RET");
+  Value *rESP = R_READ<width>(b, X86::ESP);
```

Comment:
It works for both widths (the register to variable name mapping will map ESP to RSP in the 64-bit case). Its a lot easier to write code this way than doing a switch() at the beginning of every common function.